Sources/ElevenLabsSwift/ElevenLabsSwift.swift

```diff
@@ -252,11 +252,39 @@ public class ElevenLabsSDK {
     // MARK: - Main Conversation API
 
     /// Starts a new conversation session using WebRTC/LiveKit
+    /// 
+    /// This method initializes a real-time conversation with an ElevenLabs agent.
+    /// It handles audio session configuration, WebRTC connection setup, and
+    /// agent communication through LiveKit infrastructure.
+    /// 
     /// - Parameters:
-    ///   - config: Session configuration
-    ///   - callbacks: Callbacks for conversation events
-    ///   - clientTools: Client tools callbacks (optional)
-    /// - Returns: A started `LiveKitConversationProtocol` instance
+    ///   - config: Session configuration containing agent ID or signed URL
+    ///   - callbacks: Event handlers for conversation lifecycle and messages
+    ///   - clientTools: Optional tools that the agent can call during conversation
+    /// - Returns: A connected conversation instance conforming to `LiveKitConversationProtocol`
+    /// - Throws: `ElevenLabsError` if configuration is invalid, connection fails, or audio setup fails
+    /// 
+    /// ## Usage Example:
+    /// ```swift
+    /// let config = ElevenLabsSDK.SessionConfig(agentId: "your-agent-id")
+    /// var callbacks = ElevenLabsSDK.Callbacks()
+    /// callbacks.onConnect = { id in print("Connected: \(id)") }
+    /// callbacks.onMessage = { msg, role in print("\(role): \(msg)") }
+    /// 
+    /// let conversation = try await ElevenLabsSDK.startSession(
+    ///     config: config,
+    ///     callbacks: callbacks
+    /// )
+    /// ```
+    /// 
+    /// ## Error Handling:
+    /// - `invalidConfiguration`: Check your agent ID or signed URL
+    /// - `failedToConfigureAudioSession`: Verify microphone permissions
+    /// - `connectionFailed`: Check internet connection and try again
+    /// - `authenticationFailed`: Verify API credentials
+    /// 
+    /// - Note: Requires microphone permission (`NSMicrophoneUsageDescription` in Info.plist)
+    /// - Warning: This method must be called from a Task or async context
```

Comment:
This warning is not really necessary since the compiler ensures that

![sc_202577_113722_144](https://github.com/user-attachments/assets/f98375b9-c5c8-47a0-988e-36396874ba5a)